pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -150,3 +235,54 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 	fmt.Fprintf(out, "%s\n", newName)
 	return nil
 }
+
+func hashObject(obj runtime.Object, codec runtime.Codec) (string, error) {
+	data, err := codec.Encode(obj)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%x", md5.Sum(data)), nil
+}
+
+func addDeploymentKeyToReplicationController(oldRc *api.ReplicationController, client *client.Client, deploymentKey, namespace string) error {
+	oldHash, err := hashObject(oldRc, client.Codec)
+	if err != nil {
+		return err
+	}
+	// Update all labels to include the new hash, so they are correctly adopted
+	podList, err := client.Pods(namespace).List(labels.SelectorFromSet(oldRc.Spec.Selector), fields.Everything())
+	if err != nil {
+		return err
+	}
+	for ix := range podList.Items {
+		pod := &podList.Items[ix]
+		if pod.Labels == nil {
+			pod.Labels = map[string]string{
+				deploymentKey: oldHash,
+			}
+		} else {
+			pod.Labels[deploymentKey] = oldHash
+		}
+		_, err := client.Pods(namespace).Update(pod)
+		if err != nil {
+			// TODO: retries here?
+			return err
+		}
+	}
+	if oldRc.Spec.Selector == nil {
+		oldRc.Spec.Selector = map[string]string{}
+	}
+	if oldRc.Spec.Template.Labels == nil {
+		oldRc.Spec.Template.Labels = map[string]string{}
+	}
+	oldRc.Spec.Selector[deploymentKey] = oldHash
+	oldRc.Spec.Template.Labels[deploymentKey] = oldHash
+
+	if _, err := client.ReplicationControllers(namespace).Update(oldRc); err != nil {
+		return err
+	}
+	// Note there is still a race here, if a pod was created during the update phase.
```

Comment:
Yep, I will address this in a follow on PR, we need to go through and delete the orphans.